corehq/ex-submodules/soil/__init__.py

```diff
@@ -141,9 +141,10 @@ def _task_key(self):
     @property
     def task_id(self):
         timeout = 60 * 60 * 24
-        task_id = self.get_cache().get(self._task_key(), None)
-        # This resets the timeout whenever a task is accessed in any way
-        self.get_cache().set(self._task_key(), task_id, timeout)
+        task_id = self.get_cache().get(self._task_key())
+        if task_id is not None:
+            # This resets the timeout whenever a task in the cache is accessed in any way
+            self.get_cache().set(self._task_key(), task_id, timeout)
         return task_id
```

Comment:
Seems odd that this property resets the cache every time it's accessed - know if this is accessed multiple times in the same operation?  I wonder if this could be memoized to cut down on the resets here.

Reply:
It's also not clear to me why we need to reset the timeout every time (I added that comment recently, because this caught me way off guard).  Obviously way out of scope here, but soil could really do with a top-down refactor to get a clearer means of operation and a smaller public API.

Reply:
> soil could really do with a top-down refactor

I strongly agree.

> know if this is accessed multiple times in the same operation? I wonder if this could be memoized to cut down on the resets here.

No idea